From: Paul Mackerras <paulus@samba.org>
Subject: Bug 140326 - LTC20407- libc/math/test-fenv fails

    powerpc/64: Fix bug in setting floating-point exception mode
    
    When loading up the FPU, we were using a 'ld' (load doubleword)
    instruction to get the FP exception mode from the thread_struct,
    but it's only an int field.  This changes the ld to lwz (load
    word and zero-extend).
    
    Signed-off-by: Paul Mackerras <paulus@samba.org>


Signed-off-by: Olaf Hering <olh@suse.de>

 arch/powerpc/kernel/fpu.S |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.16-rc2-olh/arch/powerpc/kernel/fpu.S
===================================================================
--- linux-2.6.16-rc2-olh.orig/arch/powerpc/kernel/fpu.S
+++ linux-2.6.16-rc2-olh/arch/powerpc/kernel/fpu.S
@@ -66,7 +66,7 @@ _GLOBAL(load_up_fpu)
 #else
 	ld	r4,PACACURRENT(r13)
 	addi	r5,r4,THREAD		/* Get THREAD */
-	ld	r4,THREAD_FPEXC_MODE(r5)
+	lwz	r4,THREAD_FPEXC_MODE(r5)
 	ori	r12,r12,MSR_FP
 	or	r12,r12,r4
 	std	r12,_MSR(r1)
